### PR TITLE
Add PCS OPA rules

### PR DESCRIPTION
**Pull request by alvarez3-hpe**
## Summary and Scope

Add PCS OPA rules

## Issues and Related PRs

* Resolves [CASMHMS-5895](https://jira-pro.its.hpecorp.net:8443/browse/CASMHMS-5895)

### Tested on:

  * Local development environment

### Test description:

Ran unit tests

## Risks and Mitigations

Low

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [X] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by kburns-hpe — APPROVED** [review]

File: kubernetes/cray-opa/templates/policies/spire.yaml [hunk 1]
```diff
@@ -110,6 +113,11 @@ data:
             {"method": "POST", "path": `^/apis/capmc/capmc/v1/get_power_cap$`},
             {"method": "POST", "path": `^/apis/capmc/capmc/v1/get_power_cap_capabilities$`},
             {"method": "POST", "path": `^/apis/capmc/capmc/v1/set_power_cap$`},
+            # PCS - Power control - component power transitioning, power status, and power capping.
```
Comment (jeremy-duckworth) [inline kubernetes/cray-opa/templates/policies/spire.yaml:116]: Does it really need to be this open?

File: kubernetes/cray-opa/templates/policies/spire.yaml [hunk 2]
```diff
@@ -77,6 +77,9 @@ data:
           {"method": "GET",   "path": `^/apis/smd/hsm/v2/.*$`},
           {"method": "HEAD",  "path": `^/apis/smd/hsm/v2/.*$`},
 
+          #PCS -> Get power status. If desired over SMD power status
+          {"method": "GET", "path": `^/apis/power-control/v1/power-status.*$`},
```
Comment (jeremy-duckworth) [inline kubernetes/cray-opa/templates/policies/spire.yaml:81]: Can these calls be filtered by xname, etc?

**Review by jeremy-duckworth — CHANGES_REQUESTED** [review]

File: kubernetes/cray-opa/templates/policies/spire.yaml (hunk 2 above)
Comment (kburns-hpe) [reply kubernetes/cray-opa/templates/policies/spire.yaml:81]: On the same note, whose the end user of these APIs? Is this being used by an application on the system?

File: kubernetes/cray-opa/templates/policies/spire.yaml (hunk 1 above)
Comment (alvarez3-hpe) [reply kubernetes/cray-opa/templates/policies/spire.yaml:116]: There really isn't anything we're restricting for PCS. The only endpoints it has are replacements for all of the CAPMC endpoints that are already made available by the OPA policy

File: kubernetes/cray-opa/templates/policies/spire.yaml (hunk 2 above)
Comment (alvarez3-hpe) [reply kubernetes/cray-opa/templates/policies/spire.yaml:81]: Yes, for example `GET /apis/power-control/v1/power-status?xname=x0c0s0b0n0&xname=x0c0s0b0n1'

Comment (jeremy-duckworth) [reply kubernetes/cray-opa/templates/policies/spire.yaml:81]: Thanks. This needs to be added to the chart/policies, is xname filtering is enabled.

File: kubernetes/cray-opa/templates/policies/spire.yaml (hunk 1 above)
Comment (jeremy-duckworth) [reply kubernetes/cray-opa/templates/policies/spire.yaml:116]: CAPMC has explicit endpoints. The new PCS ones do not. This is in Spire, needs to be very strictly controlled by endpoint, and also needs to have xname filtering support if enabled. The `wlm` identity is already problematic, as 'one node' could impact more than itself.

Comment (alvarez3-hpe) [reply kubernetes/cray-opa/templates/policies/spire.yaml:116]: The current OPA policy for CAPMC rules allow:
- Getting and setting power capping
- Getting power state
- Turning nodes on/off/reinit

These are the only things that PCS does. Having the PCS OPA policy this open does not give the node any more power than it already had. It is just that PCS will be replacing CAPMC

File: kubernetes/cray-opa/templates/policies/spire.yaml (hunk 2 above)
Comment (alvarez3-hpe) [reply kubernetes/cray-opa/templates/policies/spire.yaml:81]: I'm not sure I understand. The above policy for SMD allows for `GET /apis/smd/hms/v2/State/Component` which gets the state of everything (no xname filtering). `GET /apis/power-control/v1/power-status` does the same. Does it really need additional filtering rules?

**Review by jeremy-duckworth — APPROVED** [review]

**Pull request merged**